statsmodels/genmod/generalized_linear_model.py

```diff
@@ -700,18 +713,82 @@ def estimate_tweedie_power(self, mu, method='brentq', low=1.01, high=5.):
         power : float
             The estimated shape or power
         """
+        if exposure is not None and not isinstance(self.family.link,
+                                                   families.links.Log):
+            raise ValueError("exposure can only be used with the log link function")
+
+        # Use fit exposure if appropriate
+        if exposure is None and hasattr(self, 'exposure'):
+            # Already logged
+            exposure = np.exp(self.exposure)
+        elif exposure is None:
+            exposure = np.ones(len(self.endog))
+
+        # Use fit freq_weights if appropriate
+        if freq_weights is None and hasattr(self, 'freq_weights'):
+            # Already logged
+            freq_weights = self.freq_weights
+        elif exposure is None:
+            freq_weights = np.ones(len(self.endog))
+
+        weights = freq_weights * exposure
```

Comment:
So I struggle with this dilemma. 

Reading the Tweedie papers, it seems to me that they assume that `endog` is a rate per exposure... so they divide `endog` by `exposure` and use that as the new `endog`. Then they have `var_weights` which they set equal to exposure. `params` gives you the same value, but as we've discussed, this gives you different `cov_params` and the like as the `scale` variable is different. 

So in the `brentq` calculation, I'm calculating chi2 the same way as we do for a typical GLM... unadjusted for exposure. However the psi_p part, I am adjusting for exposure. I think what happens if I have a record that represents a policy that was in force for 1 day and there is a loss, you get a massive, massive endog value. (A policy in force for one day with a $5,000 claim has the same rate per year as an $18.25 million loss. I think we want to assign less weight to this scenario.)

As a side note, I was able to to run this function successfully on my 2 largest datasets, and both provided values of around 1.6, which is about what the literature says they should be. Those values also match the implied values from the alpha parameter that I derived for the gamma distribution of the loss.